config/config_test.go

```diff
@@ -3,32 +3,259 @@ package config
 import (
 	"testing"
 
+	"net/url"
+	"time"
+
+	"github.com/allegro/akubra/metrics"
 	"github.com/go-yaml/yaml"
 	"github.com/stretchr/testify/assert"
 )
 
 type TestYaml struct {
-	Field YAMLURL
+	Field YAMLUrl
 }
 
-func TestYAMLURLParsingSuccessful(t *testing.T) {
+func TestYAMLUrlParsingSuccessful(t *testing.T) {
 	correct := []byte(`field: http://golang.org:80/pkg/net`)
 	testyaml := TestYaml{}
 	err := yaml.Unmarshal(correct, &testyaml)
 	assert.NoError(t, err, "Should be correct")
 }
 
-func TestYAMLURLParsingFailure(t *testing.T) {
+func TestYAMLUrlParsingFailure(t *testing.T) {
 	incorrect := []byte(`field: golang.org:80/pkg/net`)
 	testyaml := TestYaml{}
 	err := yaml.Unmarshal(incorrect, &testyaml)
 	assert.Error(t, err, "Missing protocol should return error")
 }
 
-func TestYAMLURLParsingEmpty(t *testing.T) {
-	incorrect := []byte(`field:`)
+func TestYAMLUrlParsingEmpty(t *testing.T) {
+	incorrect := []byte(`field: "1"`)
+	testyaml := YAMLUrl{}
+	err := yaml.Unmarshal(incorrect, &testyaml)
+	assert.Error(t, err, "Should not even try to parse")
+	assert.Nil(t, testyaml.URL, "Should be nil")
+}
+
+func TestListenYamlParameterValidation(t *testing.T) {
+	incorrect := []byte(`Listen: ":8080"`)
 	testyaml := TestYaml{}
 	err := yaml.Unmarshal(incorrect, &testyaml)
 	assert.NoError(t, err, "Should not even try to parse")
 	assert.Nil(t, testyaml.Field.URL, "Should be nil")
 }
+
+func TestShouldValidateListenConf(t *testing.T) {
+	testListenData := []string{"127.0.0.1:8080", ":8080", ":80"}
+
+	for _, listenValue := range testListenData {
+		testConfData := prepareYamlConfig("100MB", 31, 45, "127.0.0.1:81", listenValue, nil, "client1", []string{"dev"})
```

Comment:
shouldn't we move this preparation to separate function or class with default values and overwrite only those needed for test? otherwise any later changes will be tough